frontend/src/features/admin-form/create/builder-and-design/DeleteFieldModal/DeleteFieldModal.tsx

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useMemo } from 'react'
+import {
+  ButtonGroup,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react'
+
+import Button from '~components/Button'
+import { ModalCloseButton } from '~components/Modal'
+
+import { BASICFIELD_TO_DRAWER_META } from '../../constants'
+import { useBuilderAndDesignContext } from '../BuilderAndDesignContext'
+import { useDeleteFormField } from '../mutations/useDeleteFormField'
+import {
+  BuildFieldState,
+  stateDataSelector,
+  useBuilderAndDesignStore,
+} from '../useBuilderAndDesignStore'
+
+export const DeleteFieldModal = (): JSX.Element => {
+  const stateData = useBuilderAndDesignStore(stateDataSelector)
+  const {
+    deleteFieldModal: { isOpen, onClose },
+  } = useBuilderAndDesignContext()
+
+  const fieldTypeLabel = useMemo(() => {
+    if (stateData.state === BuildFieldState.EditingField) {
+      return BASICFIELD_TO_DRAWER_META[stateData.field.fieldType].label
+    }
+    return null
+  }, [stateData])
+
+  const { deleteFieldMutation } = useDeleteFormField()
+
+  const handleDeleteConfirmation = useCallback(() => {
+    if (stateData.state === BuildFieldState.EditingField) {
+      deleteFieldMutation.mutate(stateData.field._id, {
+        onSuccess: onClose,
+      })
+    }
+  }, [deleteFieldMutation, onClose, stateData])
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalCloseButton />
+        <ModalHeader>{`Delete ${fieldTypeLabel} field`}</ModalHeader>
```

Comment:
nit: do we need a string template here? should we use direct reference instead?: `Delete {fieldTypeLabel} field`

Reply:
[4eb2272](https://github.com/opengovsg/FormSG/pull/3694/commits/4eb227265425a4359ad945b9bb4e97189d314226)